코드의 목적: French Website에 있는 여러 Factor들을 Monthly 단위로 가져와서,

특정 반기에 premium이 큰 factor가 있는지 확인해보는 작업

Time-Series 분석의 경우 훨씬 더 deep dive 해야 하기 때문에

이 코드를 그대로 믿고 투자할 것은 아니며,

그냥 rough 한 결과 확인 용도

### 0. Import Module

In [ ]:
from google.colab import drive
import pandas as pd
import statsmodels.api as sm
import numpy as np
import os
import statsmodels.api as sm
drive.mount('/content/drive', force_remount=True)

Mounted at /content/drive


### 1. About Data

각각의 폴더에 각각의 데이터가 들어가있는 형태

출처: https://mba.tuck.dartmouth.edu/pages/faculty/ken.french/data_library.html#Research

2023년 7월 3일 스냅샷이며, 추후에 (과거 데이터의 약간의 수정 포함) 바뀔 수 있음

In [ ]:
list_dir_data = os.listdir("drive/MyDrive/Colab Notebooks/data/FF_230703/")
print(list_dir_data)

['Asia_Pacific_ex_Japan_MOM_Factor_CSV', 'Developed_ex_US_5_Factors_CSV', 'Developed_5_Factors_CSV', 'Developed_Mom_Factor_CSV', 'Emerging_5_Factors_CSV', 'Emerging_MOM_Factor_CSV', 'Europe_Mom_Factor_CSV', 'Europe_5_Factors_CSV', 'Developed_ex_US_Mom_Factor_CSV', 'Asia_Pacific_ex_Japan_5_Factors_CSV', 'North_America_Mom_Factor_CSV', 'F-F_Research_Data_5_Factors_2x3_CSV', 'Japan_Mom_Factor_CSV', 'F-F_Momentum_Factor_CSV', 'North_America_5_Factors_CSV', 'Japan_5_Factors_CSV']


### 2. Aggregate Data

In [91]:
def load_ff_data(dir_data_base): #dir_data_base가 폴더명
    dir_data = "drive/MyDrive/Colab Notebooks/data/FF_230703/"+dir_data_base+"/"
    assert len(os.listdir(dir_data))==1 #한 폴더 내에 csv 파일이 하나인지 체크
    dir_data += os.listdir(dir_data)[0]
    if "F-F_Momentum" in dir_data: df = pd.read_csv(dir_data, header=11) #F-F Momentum 만 구조가 다름.
    else: df = pd.read_csv(dir_data, header=2)
    df = df.rename(columns={"Unnamed: 0": "YYYYMM"}) #첫 컬럼은 yyyymm형태며, 앞뒤로 공백 있을 때도 있음. 또한 연간 데이터랑 섞여있기도 함.
    df = df[df["YYYYMM"].apply(lambda x: len(str(x).strip())==6)].copy() #따라서 월간 데이터만 놔두기 위한 작업
    df["YYYYMM"] = df["YYYYMM"].astype(int)
    df["YEAR"] = df["YYYYMM"]//100
    df["MONTH"] = df["YYYYMM"]%100
    df = df.drop(columns=["YYYYMM"]+[col for col in df.columns if col=="RF"]).sort_values(["YEAR","MONTH"]).reset_index(drop=True)
    assert (df.iloc[-1]["YEAR"]*12+df.iloc[-1]["MONTH"])-(df.iloc[0]["YEAR"]*12+df.iloc[0]["MONTH"]) + 1 == len(df) #비어있는 데이터 있지 않은지 간단한 체크
    df = df.set_index(["YEAR","MONTH"]).stack().reset_index()
    df.columns = ["YEAR", "MONTH", "TYPE", "VALUE"]
    df["TYPE"] += ("_"+dir_data_base) #컬럼명에 출처 파일명 붙여주기.
    return df
dfAgg = pd.concat([load_ff_data(dir_data_base) for dir_data_base in list_dir_data])

### 3. 데이터 사후처리

In [104]:
dfAgg["VALUE"] = dfAgg["VALUE"].astype(float)
dfAgg = dfAgg[dfAgg["VALUE"]>-99].copy() #FF는 -99를 Missing의 Indicator로서 사용
dfAgg["Bias"] = 1
dfAgg["January"] = 0
dfAgg.loc[dfAgg["MONTH"]==1, "January"] = 1
dfAgg["2H"] = 0
dfAgg.loc[dfAgg["MONTH"]>=7, "2H"] = 1
dfAgg["NovApr"] = 0
dfAgg.loc[dfAgg["MONTH"]>=11, "NovApr"] = 1
dfAgg.loc[dfAgg["MONTH"]<=4, "NovApr"] = 1

In [93]:
dfAgg.groupby(["TYPE"])["VALUE"].agg(["min","max","mean","std"]) #이상한 데이터는 별로 없어보임. 통상적으로 Anomaly의 크기는 Monthly 1% 이하 수준.

min    max      mean       std
TYPE                                                                        
CMA_Asia_Pacific_ex_Japan_5_Factors_CSV     -7.67   8.46  0.302759  2.434050
CMA_Developed_5_Factors_CSV                 -6.46   9.58  0.209823  1.948453
CMA_Developed_ex_US_5_Factors_CSV           -8.51   7.03  0.149570  1.776740
CMA_Emerging_5_Factors_CSV                  -8.79   6.53  0.289461  1.955942
CMA_Europe_5_Factors_CSV                    -7.30   8.77  0.127215  1.825627
CMA_F-F_Research_Data_5_Factors_2x3_CSV     -7.20   9.05  0.281029  2.080735
CMA_Japan_5_Factors_CSV                    -12.99   7.54  0.070886  2.350615
CMA_North_America_5_Factors_CSV            -10.78  14.39  0.281595  2.729172
HML_Asia_Pacific_ex_Japan_5_Factors_CSV     -8.91  23.70  0.542380  3.049941
HML_Developed_5_Factors_CSV                -10.15  12.24  0.218684  2.598487
HML_Developed_ex_US_5_Factors_CSV          -10.88  12.70  0.324456  2.348603
HML_Emerging_5_Factors_CSV                 -14.35  15.53  0.652236  2.844422
HML_Europe_5_Factors_CSV                   -11.30  12.09  0.256785  2.636156
HML_F-F_Research_Data_5_Factors_2x3_CSV    -13.95  12.75  0.279166  3.002161
HML_Japan_5_Factors_CSV                    -14.25  14.58  0.285215  3.051795
HML_North_America_5_Factors_CSV            -14.07  17.58  0.124810  3.498893
Mkt-RF_Asia_Pacific_ex_Japan_5_Factors_CSV -26.00  20.52  0.619114  5.802320
Mkt-RF_Developed_5_Factors_CSV             -19.51  13.34  0.497443  4.411057
Mkt-RF_Developed_ex_US_5_Factors_CSV       -21.11  15.04  0.344785  4.802403
Mkt-RF_Emerging_5_Factors_CSV              -27.31  18.24  0.626388  6.016587
Mkt-RF_Europe_5_Factors_CSV                -22.02  16.62  0.486000  5.005857
Mkt-RF_F-F_Research_Data_5_Factors_2x3_CSV -23.24  16.10  0.556147  4.490570
Mkt-RF_Japan_5_Factors_CSV                 -17.38  24.90  0.094481  5.555699
Mkt-RF_North_America_5_Factors_CSV         -18.41  13.33  0.690025  4.420146
Mom   _F-F_Momentum_Factor_CSV             -52.05  18.20  0.629991  4.708476
RMW_Asia_Pacific_ex_Japan_5_Factors_CSV    -12.20  11.14  0.313215  2.614994
RMW_Developed_5_Factors_CSV                 -5.75   6.20  0.359975  1.487983
RMW_Developed_ex_US_5_Factors_CSV           -4.42   4.83  0.335139  1.378889
RMW_Emerging_5_Factors_CSV                  -8.93   8.30  0.198747  1.797566
RMW_Europe_5_Factors_CSV                    -5.40   6.40  0.370380  1.620440
RMW_F-F_Research_Data_5_Factors_2x3_CSV    -18.73  13.09  0.280807  2.221630
RMW_Japan_5_Factors_CSV                     -8.09   8.79  0.109063  2.112417
RMW_North_America_5_Factors_CSV            -15.45  13.94  0.362709  2.378557
SMB_Asia_Pacific_ex_Japan_5_Factors_CSV    -11.43  13.30 -0.154025  2.821471
SMB_Developed_5_Factors_CSV                 -8.64   8.37  0.028810  1.917481
SMB_Developed_ex_US_5_Factors_CSV           -9.30   7.04  0.057595  1.998362
SMB_Emerging_5_Factors_CSV                  -7.89   8.78  0.198600  2.349002
SMB_Europe_5_Factors_CSV                    -7.33   8.83  0.037215  2.102863
SMB_F-F_Research_Data_5_Factors_2x3_CSV    -15.35  18.34  0.214631  3.026620
SMB_Japan_5_Factors_CSV                    -11.53  13.10  0.072633  3.087829
SMB_North_America_5_Factors_CSV            -13.87  16.70  0.066456  2.792894
WML_Asia_Pacific_ex_Japan_MOM_Factor_CSV   -36.77  10.88  0.810844  4.221307
WML_Developed_Mom_Factor_CSV               -24.26  17.74  0.552583  3.787503
WML_Developed_ex_US_Mom_Factor_CSV         -22.52  13.13  0.639105  3.463527
WML_Emerging_MOM_Factor_CSV                -14.84   9.08  0.807681  3.002443
WML_Europe_Mom_Factor_CSV                  -26.09  13.65  0.853529  3.916606
WML_Japan_Mom_Factor_CSV                   -19.83  14.95  0.048286  4.227864
WML_North_America_Mom_Factor_CSV           -25.00  29.32  0.543632  4.633103

### 4. 간단한 OLS
통계적으로 엄밀한 방법론은 아님

하지만 크기나 유의성들을 rough하게 보는 것에는 큰 지장 없을 것으로 보임

#### 4-1. First half와 Second Half의 차이

In [97]:
df_mdl = dfAgg.groupby(["TYPE"]).apply(lambda x: sm.OLS(x["VALUE"], x[["Bias","2H"]]).fit())
df_params = df_mdl.apply(lambda x: x.params)
df_tvalues = df_mdl.apply(lambda x: x.tvalues)
df_nobs = pd.DataFrame(df_mdl.apply(lambda x: x.nobs), columns=["nobs"])

In [98]:
df_agg = pd.merge(pd.merge(df_params, df_tvalues,
                           left_index=True, right_index=True, suffixes=("_param", "_tvalue")),
                  df_nobs, left_index=True, right_index=True)

In [99]:
df_agg

Bias_param  2H_param  Bias_tvalue  \
TYPE                                                                            
CMA_Asia_Pacific_ex_Japan_5_Factors_CSV       0.333147 -0.060622     1.918765   
CMA_Developed_5_Factors_CSV                   0.191523  0.036507     1.377942   
CMA_Developed_ex_US_5_Factors_CSV             0.215888 -0.132302     1.704467   
CMA_Emerging_5_Factors_CSV                    0.187514  0.203347     1.303959   
CMA_Europe_5_Factors_CSV                      0.163198 -0.071784     1.253338   
CMA_F-F_Research_Data_5_Factors_2x3_CSV       0.268440  0.025143     2.442774   
CMA_Japan_5_Factors_CSV                       0.103096 -0.064258     0.614871   
CMA_North_America_5_Factors_CSV               0.176599  0.209462     0.907735   
HML_Asia_Pacific_ex_Japan_5_Factors_CSV       0.736701 -0.387660     3.392827   
HML_Developed_5_Factors_CSV                   0.307817 -0.177817     1.661532   
HML_Developed_ex_US_5_Factors_CSV             0.460558 -0.271518     2.753499   
HML_Emerging_5_Factors_CSV                    0.473350  0.356895     2.372789   
HML_Europe_5_Factors_CSV                      0.331624 -0.149301     1.764133   
HML_F-F_Research_Data_5_Factors_2x3_CSV       0.370084 -0.181584     2.335121   
HML_Japan_5_Factors_CSV                       0.363147 -0.155470     1.668591   
HML_North_America_5_Factors_CSV               0.182741 -0.115569     0.732228   
Mkt-RF_Asia_Pacific_ex_Japan_5_Factors_CSV    0.620152 -0.002071     1.498228   
Mkt-RF_Developed_5_Factors_CSV                0.524112 -0.053203     1.665599   
Mkt-RF_Developed_ex_US_5_Factors_CSV          0.442893 -0.195722     1.293041   
Mkt-RF_Emerging_5_Factors_CSV                 0.662414 -0.071875     1.566749   
Mkt-RF_Europe_5_Factors_CSV                   0.426701  0.118299     1.194968   
Mkt-RF_F-F_Research_Data_5_Factors_2x3_CSV    0.570557 -0.028779     2.405717   
Mkt-RF_Japan_5_Factors_CSV                    0.399746 -0.608989     1.010141   
Mkt-RF_North_America_5_Factors_CSV            0.637411  0.104963     2.021599   
Mom   _F-F_Momentum_Factor_CSV                0.409449  0.442999     2.097497   
RMW_Asia_Pacific_ex_Japan_5_Factors_CSV       0.201929  0.222010     1.083432   
RMW_Developed_5_Factors_CSV                   0.309036  0.101621     2.913036   
RMW_Developed_ex_US_5_Factors_CSV             0.314518  0.041139     3.197759   
RMW_Emerging_5_Factors_CSV                    0.104398  0.188206     0.802697   
RMW_Europe_5_Factors_CSV                      0.373604 -0.006432     3.231922   
RMW_F-F_Research_Data_5_Factors_2x3_CSV       0.171616  0.218079     1.464381   
RMW_Japan_5_Factors_CSV                       0.145330 -0.072350     0.964541   
RMW_North_America_5_Factors_CSV               0.245381  0.234064     1.447891   
SMB_Asia_Pacific_ex_Japan_5_Factors_CSV       0.050000 -0.407020     0.249064   
SMB_Developed_5_Factors_CSV                   0.365279 -0.671239     2.712370   
SMB_Developed_ex_US_5_Factors_CSV             0.477157 -0.837006     3.423203   
SMB_Emerging_5_Factors_CSV                    0.446404 -0.494394     2.719451   
SMB_Europe_5_Factors_CSV                      0.406650 -0.737003     2.753474   
SMB_F-F_Research_Data_5_Factors_2x3_CSV       0.549220 -0.668248     3.457000   
SMB_Japan_5_Factors_CSV                       0.847513 -1.545846     3.974308   
SMB_North_America_5_Factors_CSV               0.272132 -0.410314     1.369570   
WML_Asia_Pacific_ex_Japan_MOM_Factor_CSV      0.959746 -0.300107     3.189043   
WML_Developed_Mom_Factor_CSV                  0.317462  0.473878     1.177248   
WML_Developed_ex_US_Mom_Factor_CSV            0.410711  0.460320     1.665929   
WML_Emerging_MOM_Factor_CSV                   0.708276  0.201320     3.358744   
WML_Europe_Mom_Factor_CSV                     0.586954  0.537272     2.105701   
WML_Japan_Mom_Factor_CSV                     -0.545685  1.197129    -1.827699   
WML_North_America_Mom_Factor_CSV              0.372995  0.343912     1.1292

계수의 의미: Bias는 1-6월 리턴, 2H는 7-12월 리턴과 2-6월 리턴의 차이

전체적으로 SMB의 경우 1H가 더 강하며, 2H에는 모든 케이스에 reverse됨

나머지 Anomaly의 경우 1H/2H 차이가 크게 나는 케이스는 많지 않음

WML의 경우 Japan에서는 유의하게 2H가 강하며, 나머지 국가도 2H가 (유의하지 않게) 강함

#### 4-2. First half와 Second Half의 차이 (Jaunary Effect 고려)

In [101]:
df_mdl = dfAgg.groupby(["TYPE"]).apply(lambda x: sm.OLS(x["VALUE"], x[["Bias","January","2H"]]).fit())
df_params = df_mdl.apply(lambda x: x.params)
df_tvalues = df_mdl.apply(lambda x: x.tvalues)
df_nobs = pd.DataFrame(df_mdl.apply(lambda x: x.nobs), columns=["nobs"])

In [102]:
df_agg = pd.merge(pd.merge(df_params, df_tvalues,
                           left_index=True, right_index=True, suffixes=("_param", "_tvalue")),


                  df_nobs, left_index=True, right_index=True)

In [103]:
df_agg

Bias_param  January_param  \
TYPE                                                                    
CMA_Asia_Pacific_ex_Japan_5_Factors_CSV       0.244573       0.528760   
CMA_Developed_5_Factors_CSV                   0.165122       0.157605   
CMA_Developed_ex_US_5_Factors_CSV             0.143354       0.433010   
CMA_Emerging_5_Factors_CSV                    0.182857       0.027788   
CMA_Europe_5_Factors_CSV                      0.121707       0.247687   
CMA_F-F_Research_Data_5_Factors_2x3_CSV       0.198896       0.416104   
CMA_Japan_5_Factors_CSV                      -0.065549       1.006761   
CMA_North_America_5_Factors_CSV               0.219390      -0.255451   
HML_Asia_Pacific_ex_Japan_5_Factors_CSV       0.699634       0.221275   
HML_Developed_5_Factors_CSV                   0.295854       0.071419   
HML_Developed_ex_US_5_Factors_CSV             0.370854       0.535510   
HML_Emerging_5_Factors_CSV                    0.411953       0.366577   
HML_Europe_5_Factors_CSV                      0.223963       0.642703   
HML_F-F_Research_Data_5_Factors_2x3_CSV       0.209699       0.959634   
HML_Japan_5_Factors_CSV                       0.264634       0.588093   
HML_North_America_5_Factors_CSV               0.240183      -0.342910   
Mkt-RF_Asia_Pacific_ex_Japan_5_Factors_CSV    0.727683      -0.641925   
Mkt-RF_Developed_5_Factors_CSV                0.607683      -0.498895   
Mkt-RF_Developed_ex_US_5_Factors_CSV          0.571463      -0.767524   
Mkt-RF_Emerging_5_Factors_CSV                 0.626686       0.213314   
Mkt-RF_Europe_5_Factors_CSV                   0.542378      -0.690560   
Mkt-RF_F-F_Research_Data_5_Factors_2x3_CSV    0.467224       0.618276   
Mkt-RF_Japan_5_Factors_CSV                    0.527622      -0.763380   
Mkt-RF_North_America_5_Factors_CSV            0.660732      -0.139217   
Mom   _F-F_Momentum_Factor_CSV                0.822231      -2.472438   
RMW_Asia_Pacific_ex_Japan_5_Factors_CSV       0.248049      -0.275322   
RMW_Developed_5_Factors_CSV                   0.417073      -0.644952   
RMW_Developed_ex_US_5_Factors_CSV             0.428049      -0.677746   
RMW_Emerging_5_Factors_CSV                    0.211069      -0.636694   
RMW_Europe_5_Factors_CSV                      0.509268      -0.809874   
RMW_F-F_Research_Data_5_Factors_2x3_CSV       0.412508      -1.441342   
RMW_Japan_5_Factors_CSV                       0.254695      -0.652877   
RMW_North_America_5_Factors_CSV               0.396037      -0.899370   
SMB_Asia_Pacific_ex_Japan_5_Factors_CSV       0.027683       0.133226   
SMB_Developed_5_Factors_CSV                   0.277866       0.521831   
SMB_Developed_ex_US_5_Factors_CSV             0.339695       0.820608   
SMB_Emerging_5_Factors_CSV                    0.514911      -0.409029   
SMB_Europe_5_Factors_CSV                      0.183537       1.331918   
SMB_F-F_Research_Data_5_Factors_2x3_CSV       0.317592       1.385908   
SMB_Japan_5_Factors_CSV                       0.759451       0.525700   
SMB_North_America_5_Factors_CSV               0.242805       0.175074   
WML_Asia_Pacific_ex_Japan_MOM_Factor_CSV      0.920793       0.232541   
WML_Developed_Mom_Factor_CSV                  0.407744      -0.538956   
WML_Developed_ex_US_Mom_Factor_CSV            0.454268      -0.260026   
WML_Emerging_MOM_Factor_CSV                   0.818698      -0.659286   
WML_Europe_Mom_Factor_CSV                     0.550671       0.216602   
WML_Japan_Mom_Factor_CSV                     -0.376098      -1.012387   
WML_North_America_Mom_Factor_CSV              0.542256      -1.010438   

                                            2H_param  Bias_tvalue  \
TYPE                                                                
CMA_Asia_Pacific_ex_Japan_5_Factors_CSV     0.027952     1.285718   
CMA_Developed_5_Factors_CSV                 0.062908     1.082804   
CMA_Developed_ex_US_5_Factors_CSV          -0.059768     1.033490   
CMA_Emerging_5_Factors_CSV                  0.20

계수의 의미: Bias는 2-6월 리턴, January는 1월 리턴과 2-6월 리턴의 차이, 2H는 7-12월 리턴과 2-6월 리턴의 차이

전체적으로 SMB의 경우 1H가 더 강하며, 2H에는 모든 케이스에 reverse됨

Emerging을 제외하고는 January Effect도 강하며, January Effect를 제외하고도 2H가 1H에 비해 리턴이 낮음

Mom의 경우 미국 기준으로는 1H < 2H인 이유는 January의 강한 reversal 때문이며 (-2.47% per month),

Worldwide로도 WML은 January에 마이너스인 케이스가 많으나, obs 수가 적어서인지 통계적으로 유의하지는 않음

#### 4-3. 11-4와 나머지 구간 차이 (January Effect 고려)

주식을 11월-4월에 구매하고, 나머지 기간에는 구매하지 말라는 속설이 있어서 테스트 해봄

In [106]:
df_mdl = dfAgg.groupby(["TYPE"]).apply(lambda x: sm.OLS(x["VALUE"], x[["Bias","January","NovApr"]]).fit())
df_params = df_mdl.apply(lambda x: x.params)
df_tvalues = df_mdl.apply(lambda x: x.tvalues)
df_nobs = pd.DataFrame(df_mdl.apply(lambda x: x.nobs), columns=["nobs"])

In [107]:
df_agg = pd.merge(pd.merge(df_params, df_tvalues,
                           left_index=True, right_index=True, suffixes=("_param", "_tvalue")),


                  df_nobs, left_index=True, right_index=True)

In [108]:
df_agg

Bias_param  January_param  \
TYPE                                                                    
CMA_Asia_Pacific_ex_Japan_5_Factors_CSV       0.130305       0.358788   
CMA_Developed_5_Factors_CSV                   0.147716       0.061333   
CMA_Developed_ex_US_5_Factors_CSV             0.176701       0.544545   
CMA_Emerging_5_Factors_CSV                    0.359297      -0.011226   
CMA_Europe_5_Factors_CSV                      0.197817       0.374909   
CMA_F-F_Research_Data_5_Factors_2x3_CSV       0.144429       0.237300   
CMA_Japan_5_Factors_CSV                       0.124112       1.107939   
CMA_North_America_5_Factors_CSV               0.118680      -0.575697   
HML_Asia_Pacific_ex_Japan_5_Factors_CSV       0.470964       0.368970   
HML_Developed_5_Factors_CSV                   0.028173      -0.049152   
HML_Developed_ex_US_5_Factors_CSV             0.198477       0.547879   
HML_Emerging_5_Factors_CSV                    0.449557      -0.090471   
HML_Europe_5_Factors_CSV                      0.168934       0.626970   
HML_F-F_Research_Data_5_Factors_2x3_CSV       0.075961       0.825033   
HML_Japan_5_Factors_CSV                       0.073147       0.427818   
HML_North_America_5_Factors_CSV              -0.181371      -0.638606   
Mkt-RF_Asia_Pacific_ex_Japan_5_Factors_CSV    0.032030      -1.340970   
Mkt-RF_Developed_5_Factors_CSV                0.027665      -1.027273   
Mkt-RF_Developed_ex_US_5_Factors_CSV         -0.211269      -1.312909   
Mkt-RF_Emerging_5_Factors_CSV                -0.098227      -0.608941   
Mkt-RF_Europe_5_Factors_CSV                  -0.168477      -1.542424   
Mkt-RF_F-F_Research_Data_5_Factors_2x3_CSV    0.137270       0.133967   
Mkt-RF_Japan_5_Factors_CSV                   -0.228731      -0.782182   
Mkt-RF_North_America_5_Factors_CSV            0.268934      -0.704970   
Mom   _F-F_Momentum_Factor_CSV                0.634558      -2.732670   
RMW_Asia_Pacific_ex_Japan_5_Factors_CSV       0.465482      -0.226788   
RMW_Developed_5_Factors_CSV                   0.413655      -0.641333   
RMW_Developed_ex_US_5_Factors_CSV             0.370863      -0.659152   
RMW_Emerging_5_Factors_CSV                    0.233455      -0.707813   
RMW_Europe_5_Factors_CSV                      0.390254      -0.781455   
RMW_F-F_Research_Data_5_Factors_2x3_CSV       0.396964      -1.432567   
RMW_Japan_5_Factors_CSV                       0.135736      -0.576848   
RMW_North_America_5_Factors_CSV               0.471878      -0.908909   
SMB_Asia_Pacific_ex_Japan_5_Factors_CSV      -0.037563       0.516970   
SMB_Developed_5_Factors_CSV                  -0.139695       0.723879   
SMB_Developed_ex_US_5_Factors_CSV            -0.056345       1.187212   
SMB_Emerging_5_Factors_CSV                    0.098030      -0.231353   
SMB_Europe_5_Factors_CSV                     -0.173756       1.522000   
SMB_F-F_Research_Data_5_Factors_2x3_CSV      -0.138719       1.363800   
SMB_Japan_5_Factors_CSV                      -0.002893       1.364848   
SMB_North_America_5_Factors_CSV              -0.098325       0.224970   
WML_Asia_Pacific_ex_Japan_MOM_Factor_CSV      0.927254       0.547152   
WML_Developed_Mom_Factor_CSV                  0.929948      -0.379152   
WML_Developed_ex_US_Mom_Factor_CSV            0.954974      -0.164364   
WML_Emerging_MOM_Factor_CSV                   1.213920      -0.298267   
WML_Europe_Mom_Factor_CSV                     1.318964       0.440909   
WML_Japan_Mom_Factor_CSV                      0.320725      -1.405455   
WML_North_America_Mom_Factor_CSV              0.961399      -0.725515   

                                            NovApr_param  Bias_tvalue  \
TYPE                                                                    
CMA_Asia_Pacific_ex_Japan_5_Factors_CSV         0.284241     0.751934   
CMA_Developed_5_Factors_CSV                     0.113678     1.061940   
CMA_Developed_ex_US_5_Factors_CSV              -0.144882     1.397080   
CMA_Emerging_5_Factors_CSV  

계수의 의미: Bias는 5-10월 리턴, January는 1월 리턴과 5-10월 리턴의 차이, NovApr는 5-10월 리턴과 11-12 & 2-4월 리턴의 차이

MKTRf가 11-12 & 2-4월에 가장 강한 것으로 보임

Mom의 경우는 US 기준 11-12 & 2-4월에 가장 강한 것으로 보이는데, US의 결과와 다른 시장의 결과가 매우 다름 (샘플 기간의 차이 일 수 있음)

smb는 여전히 January에 강하나, 5-10월과 11-12 & 2-4월의 차이는 inconclusive